wled00/FX_fcn.cpp

```diff
@@ -1082,6 +1082,11 @@ void WS2812FX::service() {
     // last condition ensures all solid segments are updated at the same time
     if (seg.isActive() && (nowUp > seg.next_time || _triggered || (doShow && seg.mode == FX_MODE_STATIC)))
     {
+      if (!doShow) {
+        // returning bus brightness to its original value is done here, as to not interfere with asyncronous show()
+        // TODO if it is safe, prefer to restore brightness in show()
+        busses.setBrightness(_brightness, true); // "repaint" all pixels if brightness has changed
```

Comment:
Perhaps it would be ok, to restore brightness at the start of `service()`, not in a for loop.

Reply:
It only does it in the first iteration, and this makes sure it is restored right before a segment actually wants to draw.
If it is indeed safe to call gPC/sPC immediately after show(), that is of course preferable as it is easier to maintain.